File: MTGSearch/src/androidTest/java/com/dbottillo/mtgsearchfree/model/TCGPriceTest.java
package com.dbottillo.mtgsearchfree.model;

import android.os.Parcel;

import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class TCGPriceTest {

    private TCGPrice price;

    @Before
    public void setup() {
        price = new TCGPrice();
        price.setAvgPrice("avg");
        price.setHiPrice("hi");
        price.setLowprice("low");
        price.setLink("link");
        price.setError("error");
        price.setNotFound(false);
    }

    @Test
    public void tcgPrice_ParcelableWriteRead() {
        Parcel parcel = Parcel.obtain();
        price.writeToParcel(parcel, price.describeContents());
        parcel.setDataPosition(0);

        TCGPrice createdFromParcel = TCGPrice.CREATOR.createFromParcel(parcel);
        assertThat(createdFromParcel, is(price));
    }
}

File: MTGSearch/src/androidTest/kotlin/com/dbottillo/mtgsearchfree/model/TCGPriceTest.kt
package com.dbottillo.mtgsearchfree.model

import android.os.Parcel
import android.support.test.runner.AndroidJUnit4
import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class TCGPriceTest {

    lateinit var price: TCGPrice

    @Before
    fun setup() {
        price = TCGPrice()
        price.avgPrice = "avg"
        price.hiPrice = "hi"
        price.lowprice = "low"
        price.link = "link"
        price.setError("error")
        price.isNotFound = false
    }

    @Test
    fun tcgPrice_ParcelableWriteRead() {
        val parcel = Parcel.obtain()
        price.writeToParcel(parcel, price.describeContents())
        parcel.setDataPosition(0)

        val createdFromParcel = TCGPrice.CREATOR.createFromParcel(parcel)
        assertThat(createdFromParcel, `is`(price))
    }
}